projects/plugins/jetpack/class.jetpack.php

```diff
@@ -5364,19 +5369,19 @@ public function add_token_to_login_redirect_json_api_authorization( $redirect_to
 	 */
 	public function verify_json_api_authorization_request( $environment = null ) {
 		$environment = $environment === null
-			? $_REQUEST
+			? $_REQUEST // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- would appreciate a double check here.
 			: $environment;
 
 		list( $env_token,, $env_user_id ) = explode( ':', $environment['token'] );
 		$token                            = ( new Tokens() )->get_access_token( $env_user_id, $env_token );
 		if ( ! $token || empty( $token->secret ) ) {
-			wp_die( __( 'You must connect your Jetpack plugin to WordPress.com to use this feature.', 'jetpack' ) );
+			wp_die( esc_html__( 'You must connect your Jetpack plugin to WordPress.com to use this feature.', 'jetpack' ) );
 		}
 
 		$die_error = __( 'Someone may be trying to trick you into giving them access to your site. Or it could be you just encountered a bug :).  Either way, please close this window.', 'jetpack' );
 
 		// Host has encoded the request URL, probably as a result of a bad http => https redirect.
-		if ( self::is_redirect_encoded( $_GET['redirect_to'] ) ) {
+		if ( self::is_redirect_encoded( sanitize_text_field( wp_unslash( $_GET['redirect_to'] ) ) ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidated, WordPress.Security.NonceVerification.Recommended -- no site changes, we're erroring out.
```

Comment:
This looks like it's probably a URI, which `sanitize_text_field` could easily mangle by deleting percent-encoded characters.

Reply:
2dc7e0d8c1a9999981b1cd3b2d05279a6b94fdbb